### PR TITLE
feat(Core/Scripting): Implement OnIsAffectedBySpellModCheck hook

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  Implement a new global hook called when checking if spells are affected by a specific mod
-  

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

File: src/server/game/Scripting/ScriptMgr.h [hunk 1]
```diff
@@ -1517,6 +1518,9 @@ class GlobalScript : public ScriptObject
 
     // Called before the phase for a WorldObject is set
     virtual void OnBeforeWorldObjectSetPhaseMask(WorldObject const* /*worldObject*/, uint32& /*oldPhaseMask*/, uint32& /*newPhaseMask*/, bool& /*useCombinedPhases*/, bool& /*update*/) { }
+
+    // Called when checking if a spell is affected by a mod
+    virtual bool OnIsAffectedBySpellModCheck(SpellInfo const* /*affectSpell*/, SpellInfo const* /*checkSpell*/, SpellModifier const* /*mod*/) { return true;  };
```
Comment (Kitzunu) [inline src/server/game/Scripting/ScriptMgr.h:1523]: fix space before the last curly

File: src/server/game/Scripting/ScriptMgr.h [hunk 2]
```diff
@@ -2333,6 +2337,7 @@ class ScriptMgr
     void OnAfterInitializeLockedDungeons(Player* player);
     void OnAfterUpdateEncounterState(Map* map, EncounterCreditType type, uint32 creditEntry, Unit* source, Difficulty difficulty_fixed, DungeonEncounterList const* encounters, uint32 dungeonCompleted, bool updated);
     void OnBeforeWorldObjectSetPhaseMask(WorldObject const* worldObject, uint32& oldPhaseMask, uint32& newPhaseMask, bool& useCombinedPhases, bool& update);
+    bool OnIsAffectedBySpellModCheck(SpellInfo const* affectSpell, SpellInfo const* checkSpell, SpellModifier const* mod);
```
Comment (Kitzunu) [inline src/server/game/Scripting/ScriptMgr.h:2340]: misc take, but mod -> spellMod?

Comment (Nyeriah) [reply src/server/game/Scripting/ScriptMgr.h:2340]: It reflects the function it used at: https://github.com/azerothcore/azerothcore-wotlk/pull/9903/files#diff-b85d08735de1a07454ea9d244c37ce7ac6ece09993b475d6ba455237606488cfL1275

File: src/server/game/Scripting/ScriptMgr.h [hunk 3]
```diff
@@ -1517,6 +1518,9 @@ class GlobalScript : public ScriptObject
 
     // Called before the phase for a WorldObject is set
     virtual void OnBeforeWorldObjectSetPhaseMask(WorldObject const* /*worldObject*/, uint32& /*oldPhaseMask*/, uint32& /*newPhaseMask*/, bool& /*useCombinedPhases*/, bool& /*update*/) { }
+
+    // Called when checking if a spell is affected by a mod
+    virtual bool OnIsAffectedBySpellModCheck(SpellInfo const* /*affectSpell*/, SpellInfo const* /*checkSpell*/, SpellModifier const* /*mod*/) { return true; };
```
Comment (Winfidonarleyan) [inline src/server/game/Scripting/ScriptMgr.h:1523]: CanIsAffectedBySpellModCheck

Comment (Nyeriah) [reply src/server/game/Scripting/ScriptMgr.h:1523]: Hey, I appreciate the suggestion but I was told by Yehonal that hooks should follow the format: On/After/Before + <function where its used>, so thanks but I will decline this suggestion

**Pull request merged**